### ISSUE TITLE
`arithmetic_side_effects` should trigger on `pow()`

### ISSUE THREAD

**Issue by ronnodas** [rust-lang/rust-clippy#17540]
### Summary

The `.pow()` method on integers is implicit about what happens on overflow and has each of the `checked`, `overflowing`, `saturating`, `strict` and `wrapping` variants, just like the other arithmetic operations. So it should be treated the same way in this lint. 

### Lint Name

arithmetic_side_effects

### Reproducer

I tried this code:

```rust
#![deny(clippy::arithmetic_side_effects)]

fn foo(n: u32) -> u32 {
    n.pow(20)
}
```

I expected to see this happen: The lint should trigger, just like it does for `n + 1`.

Instead, this happened: Nothing.

### Version

```text

```
